src/Sentry.Unity/SentryUnitySdk.cs

```diff
@@ -30,24 +31,11 @@ private SentryUnitySdk(SentryUnityOptions options)
 
         MainThreadData.CollectData();
 
-        // On Standalone, we disable cache dir in case multiple app instances run over the same path.
-        // Note: we cannot use a named Mutex, because Unit doesn't support it. Instead, we create a file with `FileShare.None`.
-        // https://forum.unity.com/threads/unsupported-internal-call-for-il2cpp-mutex-createmutex_internal-named-mutexes-are-not-supported.387334/
-        if (ApplicationAdapter.Instance.Platform is RuntimePlatform.WindowsPlayer && options.CacheDirectoryPath is not null)
-        {
-            try
-            {
-                unitySdk._lockFile = new FileStream(Path.Combine(options.CacheDirectoryPath, "sentry-unity.lock"), FileMode.OpenOrCreate,
-                    FileAccess.ReadWrite, FileShare.None);
-            }
-            catch (Exception ex)
-            {
-                options.DiagnosticLogger?.LogWarning("An exception was thrown while trying to " +
-                                                     "acquire a lockfile on the config directory: .NET event cache will be disabled.", ex);
-                options.CacheDirectoryPath = null;
-                options.AutoSessionTracking = false;
-            }
-        }
+        // Some integrations are controlled through a flag and opt-in. Adding these integrations late so we have the
+        // same behaviour whether we're self or manually initializing
+        HandleLateIntegrations(options);
+        HandlePlatformRestrictedOptions(options, SentryPlatformServices.UnityInfo);
```

Comment:
example where we take options and also access the static mutable public property

Reply:
btw, I'm saying "static public mutable property" a lot because it's a thing you should be aware, I didn't read it btw just got the first link: 

https://www.reddit.com/r/learnprogramming/comments/7nzlec/why_are_mutable_static_variables_bad/